atlasdb-api/src/main/java/com/palantir/atlasdb/persistent/api/PhysicalPersistentStore.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.persistent.api;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.annotation.Nonnull;
+
+import org.immutables.value.Value;
+
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+
+public interface PhysicalPersistentStore extends AutoCloseable {
+    @Value.Immutable
+    interface StoreNamespace {
+        String humanReadableName();
+        UUID uniqueName();
+    }
+
+    /**
+     * Gets the value associated with the entry specified by {@code key}.
+     *
+     * @param storeNamespace handle to the namespace from which we want to retrieve the value
+     * @param key            entry key for which we want to retrieve the value
+     * @return the {@link Optional} containing the value or empty if there is no associated value
+     * @throws SafeIllegalArgumentException when {@code storeNamespace} is a handle to a non existing namespace
+     */
+    Optional<byte[]> get(StoreNamespace storeNamespace, @Nonnull byte[] key) throws SafeIllegalArgumentException;
+
+    /**
+     * Gets the values associated with the entries specified by {@code keys}.
+     *
+     * @param storeNamespace handle to the namespace from which we want to retrieve the values
+     * @param keys           representing keys for which we want to retrieve the values
+     * @return a map from keys to values
+     */
+    Map<byte[], byte[]> multiGet(StoreNamespace storeNamespace, List<byte[]> keys);
```

Comment:
based on @felixdesouza's suggestion above, we can just overload `get`/`put` here.

Also I'd prefer to try to avoid unqualified `Map<byte[], byte[]>` as it's error-prone because byte arrays by default go by reference equality

Reply:
Changed to `ByteString`.